tests/test_client.py

```diff
@@ -0,0 +1,185 @@
+import pytest
+import testinfra
+import subprocess
+import os
+import time
+from datetime import datetime
+
+
+@pytest.fixture(scope='session')
+def client(request, foreman_version):
+    name = f"smoker-client-{request.param.replace('/', '_').replace(':', '_')}"
+
+    image_id = subprocess.check_output(['podman', 'images', '-q', '--filter', f"reference={request.param}"])
+    if not image_id:
+        subprocess.check_output(['sh', 'build_images.sh', foreman_version])
+
+    container_id = subprocess.check_output(['podman', 'run', '-v', '/dev/log:/dev/log', '--systemd=true', '-dt', request.param, '/sbin/init']).decode().strip()
+
+    # return a testinfra connection to the container
+    yield testinfra.get_host("podman://" + container_id)
+
+    subprocess.check_call(['podman', 'rm', '-f', container_id])
+
+
+def wait_for_task(task, api):
+    duration = 60
+    poll = 15
+
+    while task['state'] not in ['paused', 'stopped']:
+        duration -= poll
+        if duration <= 0:
+            break
+        time.sleep(poll)
+
+        task = api.resource('foreman_tasks').call('show', {'id': task['id']})
```

Comment:
@evgeni is this something we should provide in apypie? Perhaps it could also be done smarter in the future. For example, Foreman could give hints about the time to back off.

Could also be interesting to think about asyncio here.

Reply:
Well, apypie has no concept of tasks, so in a very clean room, I'd say: no, it doesn't belong there.

on the other hand, this looks copied from FAM's foreman_helper.py, so maybe it's time to provide a `apypie.foreman` submodule?